### PR TITLE
CI: Update installation in docs-related workflow

### PR DESCRIPTION
**Pull request by akihironitta**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## What does this PR do?
Needed for https://github.com/Lightning-AI/lightning-bolts/pull/833.

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: .github/workflows/check-docs.yml [hunk 1]
```diff
@@ -35,7 +35,7 @@ jobs:
 
       - name: Install dependencies
         run: |
-          pip install -e . -U -f https://download.pytorch.org/whl/cpu/torch_stable.html
+          pip install -e ".[dev]" -U --extra-index-url https://download.pytorch.org/whl/cpu
```
Comment (akihironitta) [inline .github/workflows/check-docs.yml:38]: doctest (almost always) requires optional dependencies in other projects.

**Comment by akihironitta** [comment]
@Borda Please have a look when you get a chance!

File: .github/workflows/check-docs.yml (hunk 1 above)
Comment (Borda) [reply .github/workflows/check-docs.yml:38]: I may not here use dev, or lat least set is an option as not all our pkgs has these extras and no always you need to install packages like `pytest` for building docs...

Comment (Borda) [reply .github/workflows/check-docs.yml:38]:
```suggestion
          pip install -e . -U --extra-index-url https://download.pytorch.org/whl/cpu
```

Comment (akihironitta) [reply .github/workflows/check-docs.yml]: @Borda I understand that `pytest`, for example, isn't necessary for doctest, but we already install `[dev]` requirements in the doctest workflow in PL and Bolts, so I think having it is safe.
- https://github.com/Lightning-AI/lightning/blob/12a061f2aaefaa9ed9ccf81ab6f378835b675a7e/.github/workflows/docs-checks.yml#L52
- https://github.com/Lightning-AI/lightning-bolts/blob/5481b6e8337f4ca1381a9f0d8688f13781ef738e/.github/workflows/docs-check.yml#L32

In 4985dab, I added a new option `pip-install-extras` with the default `[dev]`, but not sure if it'll actually be useful.

**Comment by akihironitta** [comment]
Closing this in favour of #50.

**Pull request closed without merging**